index.js

```diff
@@ -49,11 +51,16 @@ Pinger.prototype._check = function check(callback) {
         }
       }
 
-      return fail(error);
+      if (self._current_failures >= self._options.tolerance) {
+        return fail(error);
+      } else {
+        self.emit('failed', error);
```

Comment:
I think we were emitting heartbeat events under this conditions before this change, is removing that intended? If so, isn't the delay also useful in case of failures? (i.e. might reveal some clues on the failure?)

Reply:
I think it causes confusion to use `heartbeat` to dispatch a `failure` in the check 🤔 Maybe renaming it would make more sense?

Passing `delay` in failure might not be as useful IMO given that:
1. If its an error from the `ping` command itself, the time would not add any value.
2. If its a forced `timeout` then the data would not be much different that what we get in the `error` that gets built.

Reply:
I was thinking if something like ping fail after a few ms vs ping fail but after 1 sec (or whatever) might indicate that the apart from failing the db is overloaded or so and if thats add any value. If not then 👍 
Agree with the event removal, but that would be a breaking change in the semantics, likely not to affect anybody but still a breaking change... if we are aware of that you have my 👍

Reply:
I would try to refrain from making assumptions of the db status from the client. Might be red-herring.

Major bump 💃